broker/src/main/java/io/zeebe/broker/exporter/metrics/MetricsExporter.java

```diff
@@ -0,0 +1,152 @@
+        && isWorkflowInstanceRecord(record)) {
+      storeWorkflowInstanceCreation(record.getTimestamp(), recordKey);
+    } else if (currentIntent == WorkflowInstanceIntent.ELEMENT_COMPLETED
+        && isWorkflowInstanceRecord(record)) {
+      final var creationTime = workflowInstanceKeyToCreationTimeMap.remove(recordKey);
+      executionLatencyMetrics.observeWorkflowInstanceExecutionTime(
+          partitionId, creationTime, record.getTimestamp());
+    }
+  }
+
+  private void storeWorkflowInstanceCreation(final long creationTime, final long recordKey) {
+    workflowInstanceKeyToCreationTimeMap.put(recordKey, creationTime);
+    creationTimeToWorkflowInstanceKeyTreeMap.put(creationTime, recordKey);
+  }
+
+  private void handleJobRecord(
+      final Record<?> record, final int partitionId, final long recordKey) {
+    final var currentIntent = record.getIntent();
+
+    if (currentIntent == JobIntent.CREATED) {
+      storeJobCreation(record.getTimestamp(), recordKey);
+    } else if (currentIntent == JobIntent.ACTIVATED) {
+      final var creationTime = jobKeyToCreationTimeMap.get(recordKey);
+      executionLatencyMetrics.observeJobActivationTime(
+          partitionId, creationTime, record.getTimestamp());
+    } else if (currentIntent == JobIntent.COMPLETED) {
+      final var creationTime = jobKeyToCreationTimeMap.remove(recordKey);
+      executionLatencyMetrics.observeJobLifeTime(partitionId, creationTime, record.getTimestamp());
+    }
+  }
+
+  private void storeJobCreation(final long creationTime, final long recordKey) {
+    jobKeyToCreationTimeMap.put(recordKey, creationTime);
+    creationTimeToJobKeyTreeMap.put(creationTime, recordKey);
+  }
+
+  private void cleanUp() {
+    final var currentTimeMillis = System.currentTimeMillis();
+
+    final var deadTime = currentTimeMillis - TIME_TO_LIVE.toMillis();
+    clearMaps(deadTime, creationTimeToJobKeyTreeMap, jobKeyToCreationTimeMap);
+    clearMaps(
+        deadTime, creationTimeToWorkflowInstanceKeyTreeMap, workflowInstanceKeyToCreationTimeMap);
+  }
+
+  private void clearMaps(
+      final long deadTime,
+      final NavigableMap<Long, Long> timeToKeyMap,
+      final Long2LongHashMap keyToTimestampMap) {
+    final var outOfScopeInstances = timeToKeyMap.headMap(deadTime);
+
+    keyToTimestampMap.keySet().removeAll(outOfScopeInstances.values())
+    outOfScopeInstances.clear();
+  }
+
+  public static ExporterCfg defaultConfig() {
+    final ExporterCfg exporterCfg = new ExporterCfg();
+    exporterCfg.setClassName(MetricsExporter.class.getName());
+    return exporterCfg;
+  }
+
+  public static String defaultExporterId() {
+    return MetricsExporter.class.getSimpleName();
+  }
+
+  private static boolean isWorkflowInstanceRecord(final Record<?> record) {
+    final var recordKey = record.getKey();
+    final var recordValue = (WorkflowInstanceRecordValue) record.getValue();
+    return recordKey == recordValue.getWorkflowInstanceKey();
```

Comment:
Nit: Could we also just check if `recordValue.getBpmnElementType() == BpmnElementType.PROCESS`? I think this is how we do it in `WorkflowInstanceMetrics`

Reply:
Ah yes thanks always forget this elementType thing :sweat_smile: